### PR TITLE
PCHR-1968: Modify Contact.getleavemanagees API

### PR DESCRIPTION
**Pull request by tunbola**
This PR modifies the Contact.getleavemanagees API.

A new required parameter: 'managed_by' is added.  Previously the API returns contacts information for contacts managed by the currently logged in user. The API has been modified to now return contact information for contacts managed by the given contactID passed in via the managed_by parameter.

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php [hunk 1]
```diff
@@ -83,21 +83,53 @@ public function testGetLeaveManageesDoesNotReturnFilteredOutFieldsWhenFilteredOu
...
 
-    //only the two contacts who are managed by the current logged in manager are returned
     $this->assertEquals(2, $result['count']);
     $this->assertEquals($result['values'][$this->contact1['id']]['id'], $this->contact1['id']);
     $this->assertEquals($result['values'][$this->contact1['id']]['display_name'], $this->contact1['display_name']);
     $this->assertEquals($result['values'][$this->contact2['id']]['id'], $this->contact2['id']);
     $this->assertEquals($result['values'][$this->contact2['id']]['display_name'], $this->contact2['display_name']);
   }
+
+  public function testGetLeaveManageesCanBeFilteredByDifferentFields() {
```
Comment (davialexandre) [inline uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php]: can we also have a test where we search for a partial name? Like, for example, searching for "civi" and then checking that it includes civihr_staff

File: uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php [hunk 2]
```diff
@@ -83,21 +83,53 @@ public function testGetLeaveManageesDoesNotReturnFilteredOutFieldsWhenFilteredOu
...
-    $result = civicrm_api3('Contact', 'getleavemanagees');
+    $result = civicrm_api3('Contact', 'getleavemanagees', ['managed_by' => $this->manager['id']]);
 
-    //only the two contacts who are managed by the current logged in manager are returned
     $this->assertEquals(2, $result['count']);
     $this->assertEquals($result['values'][$this->contact1['id']]['id'], $this->contact1['id']);
     $this->assertEquals($result['values'][$this->contact1['id']]['display_name'], $this->contact1['display_name']);
     $this->assertEquals($result['values'][$this->contact2['id']]['id'], $this->contact2['id']);
     $this->assertEquals($result['values'][$this->contact2['id']]['display_name'], $this->contact2['display_name']);
   }
```
Comment (davialexandre) [inline uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php]: can you also add a security test? If I'am logged in as manager 1 but then I call the API passing the manager 2 ID I should get an empty response (Because I should not have permission to see other manager's managees)
